### DISCUSSION TITLE
Details about ThreadDataLoader,CacheDataset,SmartCachedataset

### DISCUSSION THREAD

**Discussion by lanslotttTT** [Project-MONAI/MONAI#3863] [Q&A]
Monai is really helpful for AI research in medical images, I was a little confused when I used `ThreadDataLoader`,`CacheDataset`,`SmartCachedataset`.My question is that:
1:How `ThreadDataLoader` it works? The parameter `buffe_size` is the size of the wait queue?  what is the parameter `repeats`
used for? Why repeats data in this dataloader ？
2:  `CacheDataset` ，the parameter `num_worker` is related to my cpu cores? How do I specify `cache_rate` in order to not fill my memory?
3:Does I need update data before every traning step when using `SmartCachedataset`? Since it only have a part of dataset.

**Comment by Nic-Ma** [comment]
Hi @lanslotttTT ,

Thanks for your interest here.
For your 3 questions, you can find the details in the doc-string:
1. https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/thread_buffer.py#L82
2. https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/dataset.py#L690, you need to try suitable cache_rate.
3. https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/dataset.py#L836

Thanks.
